### PR TITLE
Verify appeal emails

### PR DESCRIPTION
**Pull request by pezholio**
This adds a new helper to verify that an email has been sent via GOV.UK Notify, we then use it in the appeal specs, so we can ensure that email(s) have been sent as expected. There's a couple of new config vars that need to be set in the various pipelines, I've added them here, but still need to:

- [x] Add config to the UI e2es
- [x] Add config to the API e2es

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Add a `.env` file to the root of the project with the following variables:
 ```text
 HMPPS_AUTH_USERNAME= # A valid HMPPS Auth Username
 HMPPS_AUTH_PASSWORD= # A valid HMPPS Auth Password
+NOTIFY_API_KEY= # An API key for GOV.UK Notify
```
Comment (yndajas) [inline README.md:29]: Is the answer to "How do I get a/the `NOTIFY_API_KEY`?" just "ask someone/in Slack"?

File: steps/email.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { isAfter, parseISO, subMinutes } from 'date-fns'
+import { NotifyClient } from 'notifications-node-client'
+import { expect } from '@playwright/test'
+
+export const verifyEmailSent = async (subject: string, emailAddress: string) => {
```
Comment (yndajas) [inline steps/email.ts:5]: `emailAddress: TestOptions['user']['email']` might be possible; not sure if there's another type for the subject parameter

File: steps/email.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const client = new NotifyClient(process.env.NOTIFY_API_KEY)
+
+  const response = await client.getNotifications('email')
+  const { notifications } = response.data
+
+  const matchedNotification = notifications.find(
+    notification =>
+      isAfter(parseISO(notification.sent_at), subMinutes(Date.now(), 1)) &&
```
Comment (yndajas) [inline steps/email.ts:13]: Would it be worth abstracting this into a separate method for readability? Is it essentially `notificationWasSentInTheLastMinute`?

File: steps/apply.ts [hunk 4]
```diff
@@ -513,7 +513,11 @@ export const withdrawAnApplicationAfterSubmission = async (page: Page, applicati
   await expect(page.getByRole('alert', { name: 'Success' })).toContainText('Success')
 }
 
-export const recordAnAppealOnApplication = async (page: Page, applicationId: string) => {
+export const recordAnAppealOnApplication = async (
+  page: Page,
+  applicationId: string,
+  decision: 'Upheld' | 'Rejected',
```
Comment (yndajas) [inline steps/apply.ts:519]: Looks like `'Upheld' | 'Rejected'` is used in a few places now - is there a suitable place to abstract this into a named type?

**Review by yndajas — COMMENTED** [review]
In principle this looks good to me, but it looks like something's failing at the moment?

File: README.md (hunk 1 above)
Comment (pezholio) [reply README.md:29]: Yeah, I reckon so, we don't have a single place for this stuff (yet)

**Comment by pezholio** [comment]
@yndajas I think the fail is related to the fix @richpjames is attempting in #82 - will wait until that's in

**Review by yndajas — APPROVED** [review]

**Pull request merged**